spring-cloud-common-security-config-core/src/test/java/org/springframework/cloud/common/security/support/OAuth2TokenUtilsServiceTests.java

```diff
@@ -0,0 +1,186 @@
+ * limitations under the License.
+ */
+package org.springframework.cloud.common.security.support;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.time.Instant;
+
+import org.junit.Test;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.OAuth2AccessToken.TokenType;
+
+/**
+ *
+ * @author Gunnar Hillert
+ *
+ */
+public class OAuth2TokenUtilsServiceTests {
+
+	@Test
+	public void testGetAccessTokenOfAuthenticatedUserWithNoAuthentication() {
+		SecurityContextHolder.getContext().setAuthentication(null);
+
+		final OAuth2AuthorizedClientService oauth2AuthorizedClientService = mock(OAuth2AuthorizedClientService.class);
+		OAuth2TokenUtilsService oAuth2TokenUtilsService = new DefaultOAuth2TokenUtilsService(oauth2AuthorizedClientService);
+
+		try {
+			oAuth2TokenUtilsService.getAccessTokenOfAuthenticatedUser();
+		}
+		catch (IllegalStateException e) {
+			assertEquals(
+				"Cannot retrieve the authentication object from the SecurityContext. Are you authenticated?",
+				e.getMessage());
+			return;
+		}
+		fail("Expected an IllegalStateException to be thrown.");
+	}
+
+	@Test
+	public void testGetAccessTokenOfAuthenticatedUserWithWrongAuthentication() {
+		final Authentication authentication = mock(Authentication.class);
+		SecurityContextHolder.getContext().setAuthentication(authentication);
+
+		final OAuth2AuthorizedClientService oauth2AuthorizedClientService = mock(OAuth2AuthorizedClientService.class);
+		OAuth2TokenUtilsService oAuth2TokenUtilsService = new DefaultOAuth2TokenUtilsService(oauth2AuthorizedClientService);
+
+		try {
+			oAuth2TokenUtilsService.getAccessTokenOfAuthenticatedUser();
+		}
+		catch (IllegalStateException e) {
+			assertEquals(
+				"Authentication object is not of type OAuth2AuthenticationToken.",
+				e.getMessage());
+			SecurityContextHolder.getContext().setAuthentication(null);
+			return;
+		}
+		fail("Expected an IllegalStateException to be thrown.");
+		SecurityContextHolder.getContext().setAuthentication(null);
+	}
+
+	@Test
+	public void testGetAccessTokenOfAuthenticatedUserWithEmptyPrincipalName() {
+		final OAuth2AuthenticationToken authentication = mock(OAuth2AuthenticationToken.class);
+		when(authentication.getName()).thenReturn("");
+		when(authentication.getAuthorizedClientRegistrationId()).thenReturn("uaa");
```

Comment:
We can use the `uaa` as a constant as this is being used in many places.

Reply:
String `"uaa"` is now handled differently outside tests